krummstab/summaries.py

```diff
@@ -0,0 +1,395 @@
+        add_student_name(workbook, worksheet, row, email_to_name[email])
+        add_student_marks_worksheet_points_per_sheet(workbook, worksheet, _the_config,
+                                                     row, email, students_marks,
+                                                     all_sheet_names, graded_sheet_names,
+                                                     points_per_sheet_cell_addresses)
+        add_student_average(workbook, worksheet, row, graded_sheet_names, all_sheet_names)
+        add_average_conditional_formatting(workbook, worksheet, row)
+        add_plagiarism_conditional_formatting(workbook, worksheet, row, all_sheet_names)
+        add_pass_or_fail_conditional_formatting(workbook, worksheet, row, graded_sheet_names, all_sheet_names)
+    add_legend(workbook, worksheet)
+    worksheet.autofit()
+
+
+def load_marks_files(marks_dir: Path, _the_config: config.Config):
+    """
+    Loads the data of the individual marks files in the specified directory.
+    """
+    marks_files = marks_dir.glob("points_*_individual.json")
+    if _the_config.points_per == 'exercise':
+        students_marks = defaultdict(lambda: defaultdict(dict))
+        graded_sheet_names = defaultdict(list)
+    else:
+        students_marks = defaultdict(dict)
+        graded_sheet_names = set()
+    tutors = defaultdict(set)
+    for file in marks_files:
+        with open(file, 'r') as f:
+            data = json.load(f)
+            sheet_name = data["adam_sheet_name"]
+            marks = data["marks"]
+            tutors[sheet_name].add(data["tutor_name"])
+            if _the_config.points_per == 'exercise':
+                for email, exercises in marks.items():
+                    for exercise, mark in exercises.items():
+                        if students_marks[email][sheet_name].get(exercise):
+                            logging.warning(f'{exercise} of sheet {sheet_name} is marked multiple times for {email}!')
+                        students_marks[email][sheet_name][exercise] = mark
+                        if exercise not in graded_sheet_names[sheet_name]:
+                            graded_sheet_names[sheet_name].append(exercise)
+            else:
+                graded_sheet_names.add(sheet_name)
+                for email, mark in marks.items():
+                    if students_marks[email].get(sheet_name):
+                        logging.warning(f'Sheet {sheet_name} is marked multiple times for {email}!')
+                    students_marks[email][sheet_name] = mark
+    for sheet_name, tutor_list in tutors.items():
+        for tutor in _the_config.classes if _the_config.marking_mode == 'static' else _the_config.tutor_list:
+            if tutor not in tutor_list:
+                logging.warning(f'There is no file from tutor {tutor} for sheet {sheet_name}!')
+    return students_marks, graded_sheet_names
+
+
+def create_email_to_name_dict(_the_config: config.Config) -> dict[str, tuple[str, str]]:
```

Comment:
We could consider moving this function to the file dealing with teams once it exists as the function is generic and may be useful elsewhere too.